paasta_tools/drain_lib.py

```diff
@@ -149,12 +149,15 @@ class CrashyDrainDrainMethod(NoopDrainMethod):
     async def drain(self, task: DrainTask) -> None:
         raise Exception("Intentionally crashing for testing purposes")
 
-
-@register_drain_method('crashy_is_safe_to_kill')
-class CrashySafeToKillDrainMethod(NoopDrainMethod):
     async def is_safe_to_kill(self, task: DrainTask) -> bool:
         raise Exception("Intentionally crashing for testing purposes")
 
+    async def stop_draining(self, task: DrainTask) -> None:
+        raise Exception("Intentionally crashing for testing purposes")
+
+    async def is_draining(self, task: DrainTask) -> bool:
+        raise Exception("Intentionally crashing for testing purposes")
+
```

Comment:
I think my reading of setup_marathon_job is that these four methods would all get called -- but I'm not totally sure. Is it worth counting calls and making sure?

Reply:
Is it possible to count calls with behave? I looked into counting calls in the unit tests, but there's no test case that calls all the methods.

Reply:
behave doesn't provide any functionality to do it AFAIK, but we could make the methods increment some class variables or something.

Reply:
Hm, I don't think it's worth it. The important behavior is that the bounce works even if the methods crash. I did verify that they're being called now, and that the test fails, if you break their exception handlers.